generator.go

```diff
@@ -129,6 +133,20 @@ func (g *DataGenerator) generateInternal(schema *spec.Schema, requestPath string
 		return nil, nil
 	}
 
+	// If we replaced a primary object ID, then also replace any of values that
+	// we happen to find which had the same value as it. These will usually be
+	// IDs in child objects that reference the parent.
+	//
+	// For example, a charge may contain a sublist of refunds. If we replaced
+	// the charge's ID, we also want to replace that charge ID in every one of
+	// the child refunds.
+	if replacedID != nil && schema.Type == "string" {
+		valStr, ok := example.value.(string)
+		if ok && valStr == *replacedID {
+			example = &Example{value: *id}
+		}
+	}
```

Comment:
What if we did this calculation in the fixture generation process instead? For example, we could set the `id` field's value in the fixture to something like `$id`, and then `stripe-mock` could just replace the string `$id` with the actual ID wherever it appears. I think this would be more robust and flexible.

Reply:
Hm, definitely an interesting idea, but wouldn't we just be moving this logic from one place to another? I'm not sure that doing it from within the API's generator will make the process all that much easier.

I'm also kind of a fan that today the fixtures really are just fixtures — i.e., we haven't developed any specialized pseudo-format that a consumer needs to worry about pre-processing before they're usable. I don't feel quite strongly enough about that to say that we should never do it, but I'd generally prefer to delay until we're really sure that it's something we want.